src/base/system.h

```
@@ -937,6 +937,27 @@ int net_addr_comp_noport(const NETADDR *a, const NETADDR *b);
  */
 void net_addr_str(const NETADDR *addr, char *string, int max_length, int add_port);
 
+/**
+ * Turns url string into a network address struct.
+ * The url format is tw-0.6+udp://{ipaddr}[:{port}]
+ * ipaddr: can be ipv4 or ipv6
+ * port: is a optional internet protocol port
+ *
+ * This format is used by the ddnet master server.
+ *
+ * Examples:
+ *   tw-0.6+udp://127.0.0.1
+ *   tw-0.6+udp://127.0.0.1:8303
+ *
+ * @param addr Address to fill in.
+ * @param string String to parse.
+ * @param pHost	buffer to write the host to (including the port if given)
+ * @param host_size size of the host buffer
+ *
+ * @return 0 on success
```

Since you rely on `-1` as a return value, it should be documented. Perhaps switch to `1` if the address can't be parsed and `<0` if there's another error. This way, you can also return an error if the address can't be parsed, but no host buffer has been passed in.